src/i18n/scripts/intl-imports.js

```diff
@@ -0,0 +1,116 @@
+
+async function generateDirectoryIndex(directory) {
+    console.log('generateDirectoryIndex', directory);
+
+    const importLines = [];
+
+    const messagesLines = [];
+
+    try {
+        const files = await fs.readdir(`${messagesDir}/${directory}`, { withFileTypes: true });
+
+        const jsonFiles = files.filter(file => file.isFile() && file.name.endsWith('.json'))
+
+        if (!jsonFiles.length) {
+            console.error(`Skipping ${directory} because no .json translation files were found.`);
+            return;
+        }
+
+        for (const [i, file] of jsonFiles.entries()) {
+            const languageCode = file.name.replace(/\.json$/, '');
+            const languageCodeDash = languageCode.toLowerCase().replace(/_/g, '-')
+            const uniqueMessageVariable = `messages${i}${languageCode}`;
+            const filePath = `${messagesDir}/${directory}/${languageCode}.json`
+
+            try {
+                const entries = JSON.parse(await fs.readFile(filePath, { encoding: 'utf8' }));
+
+                if (!Object.keys(entries).length) {
+                    console.error(`Skipping ${directory}'s ${languageCode} due to empty translation files were found.`);
+                    continue;
+                }
+            } catch (e) {
+                console.error(`Skipping ${directory}'s ${languageCode} due to error: ${e}.`);
+                continue;
+            }
+
+            console.log(languageCode);
+
+            importLines.push(
+                `import ${uniqueMessageVariable} from './${languageCode}.json';`
+            );
+
+            messagesLines.splice(1, 0, `  '${languageCodeDash}': ${uniqueMessageVariable},`)
+        }
+
+        if (importLines.length) {
+            const messagesFileContent = [
+                '// This file is generated by the Open edX generate-mfe-i18n-imports.',
+                '// Refer to docs yada yada ...',
+                '//\n',
+                importLines.join('\n'),
+                'const messages = {',
+                messagesLines.join('\n'),
+                '};',
+                'export default messages;',
+                '',
+            ].join('\n');
+
+            await fs.writeFile(`${messagesDir}/${directory}/index.js`, messagesFileContent);
+
+            console.log(messagesFileContent)
+        } else {
+            console.error(`Skipping ${directory} because no languages were found.`);
+        }
+    } catch (e) {
+        console.error(`Skipping ${directory} due to error: ${e}.`);
+    }
+}
+
+async function generateMainIndex(directories) {
+    console.log('generateMainIndex');
+
+    const importLines = [];
+    const exportLines = [];
+
+    for (const [i, directory] of directories.entries()) {
+        try {
+            await fs.access(`${messagesDir}/${directory}/index.js`, fs.constants.R_OK);
+            const directoryJsName = directory.replace(/[^a-z]/ig, '');
+
+            const moduleName = `messages${i}${directoryJsName}`;
```

Comment:
Another spot where we're just adding a number for uniqueness. I'm not completely opposed to this pattern, I just feel it leads to names that lack meaning.

Any ideas about how to avoid adding numbers that don't have any meaning would be awesome!

Reply:
Ditto. Camel case is now used.